lib/services/activities.ts

```diff
@@ -3,7 +3,60 @@ import { throwForbidden }                                from "../utils/errors"
 import { ActivitiesApi }                                 from "sunshine-conversations-client"
 import { abilities }                                     from "./abilities"
 import config                                            from "../config"
-import db                                                from "../db"
+import { sql }                                           from "../db"
+import { isFreshRecord, pubsub }                         from "./events"
+
+const transformToPrismaDAO = (obj: any): ReadReceipt => {
+  return {
+    ...obj,
+    createdAt: new Date(obj.createdAt),
+    updatedAt: new Date(obj.updatedAt)
+  }
+}
+
+/**
+ * this atomic query upserts ReadReceipt with the latest message of a conversation
+ * 
+ * If there are no messages for a particular conversationId, 
+ * the insert clause would simply insert nothing
+ */
+async function createOrUpdateReadReceipt(
+  conversationId: number, 
+  staffId: number, 
+  authorType: string
+): Promise<ReadReceipt|null> {
+  const result = await sql`
+    insert into "ReadReceipt" ("userId", "userType", "conversationId", "lastReadMessageId", "updatedAt") 
+    select ${staffId}, ${authorType}, ${conversationId}, "Message"."id", now()
```

Comment:
are we safe from SQL injection? Otherwise all good to go

Reply:
@steveyken https://github.com/crossroads/api.goodchat/pull/36#discussion_r669219050